### PR TITLE
Change to UEFI-supporting instance-type

### LINKED ISSUE #661: Update Launch-Test Instance-Type to One That Supports EFI
### Expected behavior

Test-launches work whether or not the bootstrap-AMI is EFI-enabled or not

### Actual behavior

With the use of `m4.large` instance types in the current [tests/minimal-linux.pkr.hcl](https://github.com/plus3it/spel/blob/245709b676f2f5456291348c2cc5e5e325a72d99/tests/minimal-linux.pkr.hcl#L78), test launches will fail if the bootstrap AMI was EFI-enabled.

### Steps to reproduce behavior

1. Create EFI-enabled AMI using EFI-enabled bootstrap-AMI
2. Execute spel launch-test using an instance type that doesn't support EFI-enabled
3. Wait for test to experience a timeout-failure
4. Look at failed test-EC2's console screencap and find a failed boot like:
    image
 

### Context/Specifications

Things like:
- Template name
- AWS Component
- OS/AMI information

### Suggestions for fix

Update instance-type to one that supports EFI. Using:

```bash
aws ec2 describe-instance-types \
  --filters Name=supported-boot-mode,Values=uefi 'Name=instance-type,Values=t*' \
  --query 'InstanceTypes[?
    ProcessorInfo.SupportedArchitectures[] == `["x86_64"]`
  ].[InstanceType]' --output text | \
sort
```
Any of the following instance-types:
```bash
t3.2xlarge
t3.large
t3.medium
t3.micro
t3.nano
t3.small
t3.xlarge
t3a.2xlarge
t3a.large
t3a.medium
t3a.micro
t3a.nano
t3a.small
t3a.xlarge
```
Should suffice. Alternately, if one wishes to stay with the `m` family of instance-types, upgrade to `m5`, `m6i` or `m7i`.

### PR DESCRIPTION
**Pull request by ferricoxide**
Update test-launch EC2 to a UEFI-supporting instance-type

Closes #661

### CODE REVIEW

File: tests/scripts/grow_check.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+elif [[ ${ROOT_DSKPRT} == /dev/xvd* ]]
+then
+  echo "Xen Virtual Disk devices not supported"
+  exit 1
+fi
+
+SEL_MODE="$( getenforce )"
+
+# Run the grow-part task
+[[ -d /sys/fs/selinux ]] && setenforce Permissive
```
Comment (lorengordon) [inline tests/scripts/grow_check.sh:36]: shellcheck sees all...

```suggestion
[[ -d /sys/fs/selinux ]] && setenforce Permissive
```

Comment (ferricoxide) [reply tests/scripts/grow_check.sh]: This was marked "draft" for a reason. ;)

**Comment by ferricoxide** [comment]
Forgot to trim the build-target list on the test-run, so accidentally exercised everything…

~~~text
    amazon-ebs.minimal-centos-7-hvm: Attempting to grow /dev/nvme0n1p2... + growpart /dev/nvme0n1 2
    amazon-ebs.minimal-rhel-7-hvm: Attempting to grow /dev/nvme0n1p2... + growpart /dev/nvme0n1 2
    amazon-ebs.minimal-rhel-8-hvm: Attempting to grow /dev/nvme0n1p4... + growpart /dev/nvme0n1 4
    amazon-ebs.minimal-ol-8-hvm: Attempting to grow /dev/nvme0n1p2... + growpart /dev/nvme0n1 2
    amazon-ebs.minimal-centos-8stream-hvm: Attempting to grow /dev/nvme0n1p2... + growpart /dev/nvme0n1 2
~~~

It *should* also continue to work with pre-Nitro instance types (hand ran the script on an `m4.large` – the previously-used instance-type – and the script functioned as expected. Similarly, the script was tested to ensure that it aborts if the `/` filesystem is hosted on a non-LVM device.

**Comment by lorengordon** [comment]
go codebuild go

**Review by lorengordon — APPROVED** [review]

**Pull request merged**